### ISSUE TITLE
Ignore regex support

### ISSUE THREAD

**Issue by nizamsp** [fstab/grok_exporter#20]
Thanks for the library and your passion. Is there a way to skip matching some records by overriding a match rule. I keep getting lot of 404's from /phpMyAdmin/ and other bot attacks. I want to avoid creating timelines for them by ignoring them in the grok exporter.

Any way to achieve it the current grok exporter?

**Comment by fstab** [comment]
Sorry for the late reply. There is currently no configuration option to ignore specific log lines (although this might be a nice feature). However, you can construct a regular expression in your `match` using [negative lookahead](https://www.regular-expressions.info/lookaround.html):

Example: Given the following log lines:

```
2016-04-20 01:53:59 phpMyAdmin something
2016-04-20 01:53:59 valid somethingElse
2016-04-20 01:53:59 phpMyAdmix something
```

If you configure `match: %{EXIM_DATE} %{PATH:p} %{MSG:m}` it will match all three lines. But if you configure `match: '^(?!.*phpMyAdmin)%{EXIM_DATE} %{PATH:p} %{MSG:m}'` it will match only the first and third line, the second line will be omitted because it contains the string `phpMyAdmin`.

**Comment by nizamsp** [comment]
Thank you fabian. I ended up doing with regex lookahead only. Glad to see similar solution from you.
I have been able to bring up the whole prometheus stack on a 60 machine docker cluster within a week without any prior knowledge. Thanks to the prometheus community :)

**Comment by nizamsp** [comment]
There should be some way to expire/ignore some of the timeseries with lesser counts. some moron fired some random request to our websites and we are constantly keeping the count of that api path in grok and the memory used by grok exporter is around 4G now.

**Comment by fstab** [comment]
This is a good idea. I started today to implement a configurable retention time for metrics, and I am hoping to finish it in the next one or two weeks.

**Comment by fstab** [comment]
Released v0.2.3, which allows to configure a `retention` for each metric.
